### ISSUE TITLE
Sealed trait doesn't get encoded as an Avro enum

### ISSUE THREAD

**Issue by cb372** [sksamuel/avro4s#82]
The Type Mappings table in the README suggests that I can use a sealed trait hierarchy to represent an enum, but it doesn't seem to work. Am I doing something wrong?

MyEnum.scala:

```scala
package foo

import com.sksamuel.avro4s._

sealed trait MyEnum
case object Wow extends MyEnum
case object Yeah extends MyEnum

case class Container(myEnum: MyEnum)
```

REPL:

```scala> import com.sksamuel.avro4s._
import com.sksamuel.avro4s._

scala> AvroSchema[foo.MyEnum]
res1: org.apache.avro.Schema = {"type":"record","name":"MyEnum","namespace":"foo","fields":[]}

scala> AvroSchema[foo.Container]
res2: org.apache.avro.Schema = {"type":"record","name":"Container","namespace":"foo","fields":[{"name":"myEnum","type":{"type":"record","name":"MyEnum","fields":[]}}]}
```

I would expect the sealed trait to be encoded in the schema as:

```
{
  "type": "enum",
  "symbols": [ "Wow", "Yeah" ]
}
```

I thought this might be a symptom of SI-7046, but I can see that the shapeless Coproduct is derived just fine:

```
scala> Generic[foo.MyEnum]
res4: shapeless.Generic[foo.MyEnum]{type Repr = shapeless.:+:[foo.Wow.type,shapeless.:+:[foo.Yeah.type,shapeless.CNil]]} = anon$macro$3$1@57f23793
```

**Comment by cb372** [comment]
Sorry, ignore that last bit about Coproduct. Just looked at the code more closely and realised that Coproduct maps to a union, not an enum, so it's irrelevant.

Actually I can't find any code related to sealed trait hierarchies. Is this a feature that got lost somewhere along the way?

**Comment by sksamuel** [comment]
There's this.

https://github.com/sksamuel/avro4s/blob/master/avro4s-macros/src/main/scala/com/sksamuel/avro4s/SchemaFor.scala#L220

**Comment by krzemin** [comment]
Support for sealed traits was introduced with #102 and #103.

**Comment by oliverlockwood** [comment]
@sksamuel @krzemin Is there any chance of you issuing a release including this functionality?  It's frustrating to hear that it was fixed in February but there's been no new release since `1.6.4` in January 😃

**Comment by sksamuel** [comment]
Closing this issue as the fix was released a long time ago.

**Comment by edmondo1984** [comment]
In which version @sksamuel is this released ? With 1.8.3 and Scala 2.11.12 I have problems, while for example Circe derivation works fine for the same sealed trait with two children

**Comment by xuanli09** [comment]
We're also seeing this get encoded as record types with empty fields in 1.8.3
